### ISSUE TITLE
Allow hiding message link embeds

### ISSUE THREAD

**Issue by Obsidianninja11** [Vencord/plugin-requests#106]
### Plugin Description

A way to hide a single message link embed would be very useful, since sometimes they're too long and take up room in the chat, or are just useless.
Possible ways to remove it are to have a right click prompt, or an x
![image](https://github.com/Vencord/plugin-requests/assets/108832807/8842a1fb-832e-44ce-b555-9c44ffda3031)

### Request Agreement

- [X] I read the above and made sure my plugin is possible and allowed
- [X] I checked the [existing plugins](https://vencord.dev/plugins) and made sure my plugin doesn't already exist
- [X] I [searched the existing issues](https://github.com/Vencord/plugin-requests/issues?q=is%3Aissue) and made sure a similar request doesn't already exist

**Comment by Obsidianninja11** [comment]
One advantage to having it in the right click context menu is being able to show it if the user changes their mind about hiding it.
Also there's less visual change

**Comment by Obsidianninja11** [comment]
With this, there could also be an option to initially hide the embed until you un hide it with the right click context menu

**Comment by Kyuuhachi** [comment]
I have tried making this in the past (specifically, making it integrate with discord's builtin suppress embeds function), but I couldn't get that working properly. So I made a plugin that shows it in the tooltip instead: [MessageLinkTooltip](https://github.com/Kyuuhachi/VencordPlugins/tree/main/MessageLinkTooltip).

**Comment by Obsidianninja11** [comment]
> (specifically, making it integrate with discord's builtin suppress embeds function)

I'm pretty sure this is a vencord feature. The suppress button in right click context menu doesn't show unless this plugin is enabled
![2024-02-25_00 41 15](https://github.com/Vencord/plugin-requests/assets/108832807/f6abce36-ff5e-4e97-8d0b-17360767ad09)

**Comment by Kyuuhachi** [comment]
Yeah, but the X button when you hover an embed is vanilla. Vencord just makes ot a bit easier to find, and allows unsetting the flag.

Feb 25, 2024 10:43:45 Obsidian ***@***.***>:
